pytest_sentry.py

```diff
@@ -94,52 +97,52 @@ def __init__(self, *args, **kwargs):
 
 def hookwrapper(itemgetter, **kwargs):
     """
-    A version of pytest.hookimpl that sets the current hub to the correct one
+    A version of pytest.hookimpl that sets the current scope to the correct one
     and skips the hook if the integration is disabled.
 
     Assumes the function is a hookwrapper, ie yields once
     """
 
     @wrapt.decorator
-    def _with_hub(wrapped, instance, args, kwargs):
+    def _with_scope(wrapped, instance, args, kwargs):
         item = itemgetter(*args, **kwargs)
-        hub = _resolve_hub_marker_value(item.get_closest_marker("sentry_client"))
+        scope = _resolve_scope_marker_value(item.get_closest_marker("sentry_client"))
 
-        if hub.get_integration(PytestIntegration) is None:
+        if scope.client.get_integration(PytestIntegration) is None:
             yield
         else:
-            with hub:
+            with use_isolation_scope(scope):  # TODO: What do we do with scope here?
```

Comment:
we should not fork the scope here. we need to set it as current scope and unset it at the end of the with-stmt. I think `use_scope` is more correct

specifically, `sentry_test_scope` should be usable from within tests to access the transaction created by the plugin, and modify tags. I think the tags might work but the assertion probably fails

```
def test_foo(sentry_test_scope):
    assert sentry_test_scope.span is not None
    sentry_test_scope.set_tag("foo", "bar")
```

Reply:
Okay, I switched to `use_scope` and added a test with the span is not None assertion (the assertion failed with `use_isolation_scope`)